### DISCUSSION TITLE
Support HierProb3D method (Sengupta et al., ICCV 2021)

### DISCUSSION THREAD

**Discussion by kristijanbartol** [open-mmlab/mmhuman3d#230] [Ideas]
## Description

[HierProb3D](https://github.com/akashsengupta1997/HierarchicalProbabilistic3DHuman) estimates 3D human pose and **accurate** body shape from single or multiple images. The training is a bit more complicated as it's done on the synthetic data, generated on-the-fly (SMPL bodies are rendered and textured on random background, the gradients are off). For both training and inference, we need:

- [2D keypoint detector](https://github.com/akashsengupta1997/HierarchicalProbabilistic3DHuman/blob/main/models/pose2D_hrnet.py) (which is easy to import)
- [edge detector](https://github.com/akashsengupta1997/HierarchicalProbabilistic3DHuman/blob/main/models/canny_edge_detector.py).

## Questions

Where do we implement edge detector? Should we use this repository or some other Open-MMLab repo? I guess that this repository is fine, because this edge detection will be used only by this method. On the other hand, edge detection logically fits better into [semantic segmentation repo](https://github.com/open-mmlab/mmsegmentation)...

**Comment by caizhongang** [comment]
I think there are two ways:

1. provide the edge detection results as part of the data (simple, but not scalable)
2. If the edge detector is likely to be used by future methods, I would suggest including the network architecture in models/edge_detector, pretrained weights, and a script to preprocess any data in tools/

**Reply by kristijanbartol** [reply]
Thank you for the suggestions! I prefer going with the second option, because:

1. The original implementation generates training data on-the-fly, which would not be possible if the edge detections are part of the data;
2. There might be future methods that exploit edge detection (inspired by HierProb3D), although I am not aware of any other mesh regression method that uses edge detection.

**Note** that the third option - implementing edge detection outside MMHuman3D repository, is not preferred based on @caizhongang's suggestions, which I agree with.
